src/compiler/checker.ts

```diff
@@ -44675,7 +44675,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         }
                         if (isConstantVariable(symbol)) {
                             const declaration = symbol.valueDeclaration as VariableDeclaration | undefined;
-                            if (declaration && !declaration.type && declaration.initializer && (!location || declaration !== location && isBlockScopedNameDeclaredBeforeUse(declaration, location))) {
+                            if (declaration?.kind === SyntaxKind.VariableDeclaration && !declaration.type && declaration.initializer && (!location || declaration !== location && isBlockScopedNameDeclaredBeforeUse(declaration, location))) {

```

Comment:
alternatively, I could check if the parent is not a `BindingElement`

Reply:
Would that break on parameters? I guess not, since `isConstantVariable` won't be true.

Reply:
I'll also note that this feels weird because `declaration` is `VariableDeclaration` which _must_ have `kind === SyntaxKind.VariableDeclaration`.

Reply:
> I'll also note that this feels weird because declaration is VariableDeclaration which must have kind === SyntaxKind.VariableDeclaration.

Good catch! I didn't notice it. It is `VariableDeclaration` just because of the cast - so I'll make that cast more accurate in a second :)